File: springboot/src/test/java/com/hypermedicus/Test_AccountService.java
package com.hypermedicus;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.hypermedicus.model.AccountDTO;
import com.hypermedicus.payload.response.AccountResponse;
import com.hypermedicus.payload.response.ApiResponse;
import com.hypermedicus.rds.entity.AccountClassificationEntity;
import com.hypermedicus.rds.entity.AccountEntity;
import com.hypermedicus.rds.repository.AccountClassificationEntityRepository;
import com.hypermedicus.rds.repository.AccountEntityRepository;
import com.hypermedicus.service.AccountService;

import lombok.extern.log4j.Log4j2;

import java.util.NoSuchElementException;
import java.util.UUID;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest
@Log4j2
class Test_AccountService {

    @Autowired
    private AccountService accountService;

    private AccountDTO accountDTO;

    
    @Autowired
    private AccountEntityRepository accountEntityRepository;

    @Autowired
    private AccountClassificationEntityRepository accountClassificationEntityRepository;

    @BeforeEach
    void setUp() {
        accountDTO = new AccountDTO();
        accountDTO.setPassword("password");
        accountDTO.setName("Test Name");
        accountDTO.setEmail("test@email.com");
        accountDTO.setPhone("123-456-7890");
        accountDTO.setNation("Test Nation");
        accountDTO.setHospital("Test Hospital");
        accountDTO.setDepartment("Test Department");
    }

    @Test
    void testSignUpAccount() throws JsonProcessingException {
        //log.info("AccountDTO: {}", accountSignUpDTO);
        accountService.signUpAccount(accountDTO);
        UUID uuid = accountDTO.getUuid();
        AccountEntity accountEntity = accountEntityRepository.findByUuid(uuid)
        .orElseThrow(() -> new NoSuchElementException("AccountEntity not found with UUID: " + uuid));
        
        AccountClassificationEntity accountClassificationEntity = accountClassificationEntityRepository.findByUuid(uuid)
        .orElseThrow(() -> new NoSuchElementException("accountClassificationEntity not found with UUID: " + uuid));
        
        ApiResponse<AccountResponse> apiResponse = ApiResponse.ok(AccountResponse.toResponse(accountEntity, accountClassificationEntity));
        log.info(apiResponse);

        ObjectMapper objectMapper = new ObjectMapper();
        String jsonResponse = objectMapper.writeValueAsString(apiResponse);
        log.info(jsonResponse);
        jsonResponse = objectMapper.writeValueAsString(apiResponse.getData());
        log.info(jsonResponse);
    }
}

File: fastapi/test/Test_AccountService.py
import pytest
from fastapi.testclient import TestClient
from app.HyperMedicusApplication import app
from app.model.AccountDTO import AccountSignUpDTO
from app.resources.Sqlalchemy import SessionLocal
from app.service import AccountService

client = TestClient(app)

@pytest.fixture
def accountSignupDto():
    return AccountSignUpDTO(
        password="password",
        name="Test Name",
        email="test@email.com",
        phone="123-456-7890",
        nation="Test Nation",
        hospital="Test Hospital",
        department="Test Department"
    )

@pytest.fixture(autouse=True)
def db_session():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.rollback()
        db.close()

def testSignUpAccount(accountSignupDto):
    response = client.post("/api/account/signup", json=accountSignupDto.dict())
    assert response.status_code == 200
    print(response.json())